README.md
# laravel-jetstream-innertiajs-starter

Production Ready, Extensive, Beautifully Crafted, WebApp Starter Kit.

SquisUI is a full-featured stater-kit for for your next WebApp application. It's built on top of Laravel, Livewire, InertiaJS, Tailwind and DaisyUI.

![GitHub tag (latest SemVer pre-release)](https://img.shields.io/github/v/tag/squid-ui/laravel-jetstream-innertiajs-starter?include_prereleases) ![GitHub all releases](https://img.shields.io/github/downloads/squid-ui/laravel-jetstream-innertiajs-starter/total) ![GitHub commit activity](https://img.shields.io/github/commit-activity/y/squid-ui/laravel-jetstream-innertiajs-starter) ![GitHub](https://img.shields.io/github/license/squid-ui/laravel-jetstream-innertiajs-starter)

![Dashboard](public/images/squidce/squid-ce-wellcome.png "SquidUI Welcome page after succeffsulll installation")

## SquidUI Community Edition features:

* Authentication (with 2 Factor Auth support)
* User Management
* User profiles
* Teams and Groups Management
* Multi Themes Dashboard
* much more.

## Under the hood

* [Laravel](https://laravel.com/)
* [Jetstream](https://jetstream.laravel.com/)
* [InnertiaJS](https://inertiajs.com/)
* [Vue (With Composition API)](https://vuejs.org/)
* [Pinia Store](https://pinia.vuejs.org/)
* [TailwindCss](https://tailwindcss.com/)
* [DaisyUI](https://daisyui.com/)

## Sreens

<img src="public/images/squidce/squid-ce-login.png" width="50%" height="50%">

<img src="public/images/squidce/squid-ce-dashboard.png" width="50%" height="50%">

## Why use squidUI?

**Save time without sacrificing quality**

As developers, we always have to deliver reliable, low cost and easy to deploy applications within the shortest period of time.

So, if you need to do things quick with high flexibility what can you do?

You need to speed up your production without sacrificing quality.

And how can you do that?

You can start with Laravel framework...

that'll save you some time! 😃

You can choose Vue.js and Tailwind to create your UI...

Nice! You just cut a lot more time without sacrificing code standards. 🤓

But there is something more you can do... 💭

You can choose SquidUI 🦑

A Production Ready, Extensive, Beautifully Crafted, WebApp Starter Kit.

## Community and Pro Versions

### SquidUI Community MIT LICENSE (This version)

Community edition is a lightweight version, with the basic stuff you'll need in any web application. 

It is easy to use if you're familiar with Laravel and Vue.js. This version is MIT, and it will always be free. 

You are free to use it however you like.

### SquidUI Pro LICENSE  [Visit Website](https://squidui.com/)

<img src="https://squidui.com/images/squidUI.png" width="80%">

Save thousands of hours (2960 and counting 🕓) implementing the best design patters, libraries and strategies…

And use **squidUI Pro**, the create your state-of-the-art project.

Deploy beautifully themed, secure and fast webApps inhalf of the time with **SquidUI Pro** help.

All this can be yours so you can use it in all your projects for a very small price!

## Instalation

**SquidUI Community Edition** is almost ready to use, just follow this steps:

0. Pre-requisites

You need Node.js and Composer installed in order to run SquidUI.

1. Download or clone this Repo
```
git clone https://github.com/squid-ui/laravel-jetstream-innertiajs-starter.git
```

2. Install Composer Dependencies
```
composer install
```

3. Install npm Dependencies
```
npm install
```

4. Copy .env.example to .env and edit your project and database configurations to fit your development environment needs
```
cp .env.example .env
```

5. Generate your Laravel key
```
php artisan key:generate
```

6. Run migrations
```
php artisan migrate
```

7. Build NPM dependencies
```
npm run build
```

8. Start Testing server
```
php artisan serve
```

That's all! Now goto [https://localhost:8000](https://localhost:8000) and enjoy!

## Instalation on Docker (Sail) [Sail Documentation](https://laravel.com/docs/9.x/sail)

0. Pre-requisites

You need Docker installed

1. Copy .env.example to .env
```
cp .env.example .env
```

2. Start sail, all dependencies will be installed for you
```
./vendor/bin/sail up -d
```

3. Generate you key
```
./vendor/bin/sail artisan key:generate
```

4. Install Composer and Node Dependencies
```
./vendor/bin/sail composer install
```
```
./vendor/bin/sail npm install
```

5. Run migrations
./vendor/bin/sail artisan migrate:fresh
```

6. Start server 
./vendor/bin/sail npm run dev
```

## License

MIT. Please see the [license file](LICENSE.md) for more information.


CHANGELOG.md
# SquidUI Community Edition changelog

## v0.0.1

- Laravel sail scaffolding
- added stripe cli container on docker-compose.yml
- Laravel Jetstream + InertiaJS + Teams installation
- editorconfig edited
- .eslintre.cjs included
- .prettierignore included
- .styleci.yml included
- refactored ziggy import in app.js and vite.config.js
- aliases added in vite
- DaisyUI added
- tailwind-scrollbar-daisyui added
- Vite correction wmr to enable npm run dev work
- Added change theme dropdown component
- theme-change package installed
- added pinia and pinia-plugin-persistedstate packages
- added Pinia store mainly to store theme changes
- Welcome page refactored for SquidUI
- vue-feather package installed
- Added LICENSE.md file.
- Modified README.md
- Added SquidUI Community Edition Print Screens

## v0.0.2

- Missing Pinia instance added in app.js setup

## v0.0.3

- @tailwindcss/forms removed (interfering with DaisyUI)
- Refactores all tailwind forms to fit DaisyUI Specs instead of @tailwindcss/forms


vite.config.js
import { defineConfig } from "vite";
import laravel from "laravel-vite-plugin";
import vue from "@vitejs/plugin-vue";
import path from "path";

export default defineConfig({
  server: {
    hmr: {
      host: "localhost",
    },
  },
  plugins: [
    laravel({
      input: "resources/js/app.js",
      refresh: true,
    }),
    vue({
      template: {
        transformAssetUrls: {
          base: null,
          includeAbsolute: false,
        },
      },
    }),
  ],
  resolve: {
    alias: {
      "@": path.resolve(__dirname, "resources/js"),
      "@components": path.resolve(__dirname, "resources/js/Components"),
      "@composables": path.resolve(__dirname, "resources/js/Composables"),
      "@pages": path.resolve(__dirname, "resources/js/Pages"),
      "@stores": path.resolve(__dirname, "resources/js/Stores"),
      "@static": path.resolve(__dirname, "resources/static"),
      "@shop": path.resolve(__dirname, "resources/js/Shop"),
      "@mock-data": path.resolve(__dirname, "resources/js/@mock-data"),
      ziggy: path.resolve(__dirname, "vendor/tightenco/ziggy/dist/vue.m"),
    },
  },
});


resources/js/app.js
import "./bootstrap";
import "../css/app.css";

import { createApp, h } from "vue";
import { createInertiaApp } from "@inertiajs/inertia-vue3";
import { InertiaProgress } from "@inertiajs/progress";
import { resolvePageComponent } from "laravel-vite-plugin/inertia-helpers";
import { ZiggyVue } from "ziggy";

import { createPinia } from "pinia";
import piniaPluginPersistedstate from "pinia-plugin-persistedstate";
import { useMainStore } from "@stores/main.js";
import { uiColors } from "@composables/useColors";

const pinia = createPinia();
pinia.use(piniaPluginPersistedstate);

const appName =
  window.document.getElementsByTagName("title")[0]?.innerText || "Laravel";

createInertiaApp({
  title: (title) => `${title} - ${appName}`,
  resolve: (name) =>
    resolvePageComponent(
      `./Pages/${name}.vue`,
      import.meta.glob("./Pages/**/*.vue")
    ),
  setup({ el, app, props, plugin }) {
    return createApp({ render: () => h(app, props) })
      .use(pinia)
      .use(plugin)
      .use(ZiggyVue)
      .mount(el);
  },
});

InertiaProgress.init({ color: "#4B5563" });

/* Pinia Store */
const mainStore = useMainStore(pinia);
const activeTheme = localStorage.getItem("theme")
  ? localStorage.getItem("theme")
  : "light";

const themeColors = uiColors(activeTheme);
mainStore.setuiThemeColors(themeColors);


config/jetstream.php
<?php

use Laravel\Jetstream\Features;
use Laravel\Jetstream\Http\Middleware\AuthenticateSession;

return [

    /*
    |--------------------------------------------------------------------------
    | Jetstream Stack
    |--------------------------------------------------------------------------
    |
    | This configuration value informs Jetstream which "stack" you will be
    | using for your application. In general, this value is set for you
    | during installation and will not need to be changed after that.
    |
    */

    'stack' => 'inertia',

    /*
    |--------------------------------------------------------------------------
    | Jetstream Route Middleware
    |--------------------------------------------------------------------------
    |
    | Here you may specify which middleware Jetstream will assign to the routes
    | that it registers with the application. When necessary, you may modify
    | these middleware; however, this default value is usually sufficient.
    |
    */

    'middleware' => ['web'],

    'auth_session' => AuthenticateSession::class,

    /*
    |--------------------------------------------------------------------------
    | Jetstream Guard
    |--------------------------------------------------------------------------
    |
    | Here you may specify the authentication guard Jetstream will use while
    | authenticating users. This value should correspond with one of your
    | guards that is already present in your "auth" configuration file.
    |
    */

    'guard' => 'sanctum',

    /*
    |--------------------------------------------------------------------------
    | Features
    |--------------------------------------------------------------------------
    |
    | Some of Jetstream's features are optional. You may disable the features
    | by removing them from this array. You're free to only remove some of
    | these features or you can even remove all of these if you need to.
    |
    */

    'features' => [
        Features::termsAndPrivacyPolicy(),
        Features::profilePhotos(),
        // Features::api(),
        Features::teams(['invitations' => true]),
        Features::accountDeletion(),
    ],

    /*
    |--------------------------------------------------------------------------
    | Profile Photo Disk
    |--------------------------------------------------------------------------
    |
    | This configuration value determines the default disk that will be used
    | when storing profile photos for your application's users. Typically
    | this will be the "public" disk but you may adjust this if needed.
    |
    */

    'profile_photo_disk' => 'public',

];


resources/js/Stores/main.js
import { defineStore } from "pinia";

export const useMainStore = defineStore("main", {
  state: () => ({
    /* User */
    userName: null,
    userEmail: null,
    userAvatar: null,

    /* localization */
    uiLanguage: null,

    /* theme */
    uiThemeColors: null,

    /* Media Queries Matches */
    matches: null,
  }),
  actions: {
    setUser(payload) {
      if (payload.name) {
        this.userName = payload.name;
      }
      if (payload.email) {
        this.userEmail = payload.email;
      }
      if (payload.avatar) {
        this.userAvatar = payload.avatar;
      }
    },

    setUiLanguage(payload) {
      localStorage.setItem("uiLanguage", payload);
      this.uiLanguage = payload;
    },

    setuiThemeColors(payload) {
      localStorage.setItem("uithemecolors", payload);
      this.uiThemeColors = payload;
    },

    setMatches(payload) {
      localStorage.setItem("matches", payload);
      this.matches = payload;
    },
  },
});


resources/js/Composables/useColors.js
import daisyuiColors from 'daisyui/src/colors/themes'

export function uiColors(activeTheme) {
  return {
    themeName: activeTheme,
    accent: daisyuiColors['[data-theme=' + activeTheme + ']'].accent,
    base100: daisyuiColors['[data-theme=' + activeTheme + ']']["base-100"],
    base200: daisyuiColors['[data-theme=' + activeTheme + ']']["base-200"],
    base300: daisyuiColors['[data-theme=' + activeTheme + ']']["base-300"],
    baseContent: daisyuiColors['[data-theme=' + activeTheme + ']']["base-content"],
    neutral: daisyuiColors['[data-theme=' + activeTheme + ']'].neutral,
    primary: daisyuiColors['[data-theme=' + activeTheme + ']'].primary,
    secondary: daisyuiColors['[data-theme=' + activeTheme + ']'].secondary,
    success: daisyuiColors['[data-theme=' + activeTheme + ']'].success,
    error: daisyuiColors['[data-theme=' + activeTheme + ']'].error,
    info: daisyuiColors['[data-theme=' + activeTheme + ']'].info,
    warning: daisyuiColors['[data-theme=' + activeTheme + ']'].warning,
  }
}


resources/js/Composables/useThemes.js
export const themes = [
    {
      name: "🌝  light",
      id: "light",
    },
    {
      name: "🌚  dark",
      id: "dark",
    },
    {
      name: "🧁  cupcake",
      id: "cupcake",
    },
    {
      name: "🐝  bumblebee",
      id: "bumblebee",
    },
    {
      name: "✳️  Emerald",
      id: "emerald",
    },
    {
      name: "🏢  Corporate",
      id: "corporate",
    },
    {
      name: "🌃  synthwave",
      id: "synthwave",
    },
    {
      name: "👴  retro",
      id: "retro",
    },
    {
      name: "🤖  cyberpunk",
      id: "cyberpunk",
    },
    {
      name: "🌸  valentine",
      id: "valentine",
    },
    {
      name: "🎃  halloween",
      id: "halloween",
    },
    {
      name: "🌷  garden",
      id: "garden",
    },
    {
      name: "🌲  forest",
      id: "forest",
    },
    {
      name: "🐟  aqua",
      id: "aqua",
    },
    {
      name: "👓  lofi",
      id: "lofi",
    },
    {
      name: "🖍  pastel",
      id: "pastel",
    },
    {
      name: "🧚‍♀️  fantasy",
      id: "fantasy",
    },
    {
      name: "📝  Wireframe",
      id: "wireframe",
    },
    {
      name: "🏴  black",
      id: "black",
    },
    {
      name: "💎  luxury",
      id: "luxury",
    },
    {
      name: "🧛‍♂️  dracula",
      id: "dracula",
    },
    {
      name: "🖨  CMYK",
      id: "cmyk",
    },
    {
      name: "🍁  Autumn",
      id: "autumn",
    },
    {
      name: "💼  Business",
      id: "business",
    },
    {
      name: "💊  Acid",
      id: "acid",
    },
    {
      name: "🍋  Lemonade",
      id: "lemonade",
    },
    {
      name: "🌙  Night",
      id: "night",
    },
    {
      name: "☕️  Coffee",
      id: "coffee",
    },
    {
      name: "❄️  Winter",
      id: "winter",
    },
  ]


tailwind.config.js
const defaultTheme = require("tailwindcss/defaultTheme");
const plugin = require("tailwindcss/plugin");

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./vendor/laravel/framework/src/Illuminate/Pagination/resources/views/*.blade.php",
    "./vendor/laravel/jetstream/**/*.blade.php",
    "./storage/framework/views/*.php",
    "./resources/views/**/*.blade.php",
    "./resources/js/**/*.vue",
    "./resources/js/**/*.js",
    "./resources/markdown/**/*.md",
  ],
  darkMode: "class",
  safelist: [
    "mr-1",
    "opacity-20",
    "hover:opacity-60",
    "text-base",
    "font-bold",
    "inline-block",
    "align-middle",
    "relative",
    "-mt-1",
  ],
  theme: {
    extend: {
      fontFamily: {
        sans: ["Nunito", ...defaultTheme.fontFamily.sans],
      },
    },
  },
  plugins: [
    require("@tailwindcss/forms"),
    require("daisyui"),
    require("tailwind-scrollbar-daisyui"),
    plugin(function ({ matchUtilities, theme }) {
      matchUtilities(
        {
          "aside-scrollbars": (value) => {
            const track = value === "light" ? "50" : "900";
            const thumb = value === "light" ? "300" : "600";
            const color = value === "light" ? "gray" : value;

            return {
              scrollbarColor: `${theme(`colors.${color}.${thumb}`)} ${theme(
                `colors.${color}.${track}`
              )}`,
              "&::-webkit-scrollbar-track": {
                backgroundColor: theme(`colors.${color}.${track}`),
              },
              "&::-webkit-scrollbar-thumb": {
                backgroundColor: theme(`colors.${color}.${thumb}`),
              },
            };
          },
        },
        { values: theme("asideScrollbars") }
      );
    }),
  ],
  variants: {
    scrollbar: ["dark"],
  },
  daisyui: {
    styled: true,
    themes: true,
    base: true,
    utils: true,
    logs: true,
    rtl: false,
    prefix: "",
    darkTheme: "dark",
  },
};
